### ISSUE TITLE
Empty transition-duration results in no popover events available

### ISSUE THREAD

**Issue by cont-kolomeets** [Esri/calcite-components#6582]
### Actual Behavior

In some browsers, especially Edge, popover doesn't emit events. This is observed on some Windows machines in our team. Looks like it appears mostly on slower machines. Please try the sample provided below on different Windows machines.

Looks like the root of the problem is that internally Calcite listens to "transitionstart/end" events on the "calcite-floating-ui-anim" element. But the `transition-duration` is not specified. Looks like some browsers play hide/show animation instantly and don't even through "transitionstart/end" native events. We hacked this problem by setting `transition-duration` to 100ms.

![image](https://user-images.githubusercontent.com/5318527/224013476-6c0315b2-666c-4963-af2a-e5d0a552f619.png)

![image](https://user-images.githubusercontent.com/5318527/224013517-63363963-4929-44c8-8f3a-66a3927cdb24.png)

### Expected Behavior

Popover must emit events and `transition-duration` must be specified explicitly.

### Reproduction Sample

https://codepen.io/cont-kolomeets/pen/QWVOvOR?editors=100

### Reproduction Steps

See the sample above.

### Reproduction Version

1.0.7

#### Machine 1:
Windows 10
Chrome 110.0.5481.178 - not reproducible
Edge 110.0.1587.63 - issue is observed

#### Machine 2
Windows 10
Chrome 110.0.5481.77 - issue is observed
Edge 109.0.1518.78 - issue is observed
FF 109.0.1 - issue is observed

### Regression?

Looks like a regression from 0.98

### Priority impact

p0 - emergency

### Esri team

ArcGIS Business/Community Analyst

**Comment by jcfranco** [comment]
Doing some preliminary digging into this one. ⛏️👷‍♂️

**Comment by jcfranco** [comment]
### TL;DR

I've submitted #6605 with a fix while #6604 is worked on.

### Findings

* the issue is present whenever `prefers-reduced-motion` is enabled
  * [the following](https://codepen.io/jcfranco/pen/gOdeVpq?editors=1000) could be used as a workaround
* it looks like the utils for `openCloseComponent` were never updated to handle the case where duration is set to 0s. https://github.com/Esri/calcite-components/issues/5206 addressed a special case for modal, but all open/close components will share the same problem. 
* the regression was likely introduced [here](https://github.com/Esri/calcite-components/pull/5981/files) where a workaround for `openCloseComponent` working w/ `prefers-reduced-motion` was removed

#### `openCloseComponent` breakdown

This module has the following utils:

* `connectOpenCloseComponent` – called on `connectedCallback` and adds listeners for transition start/end on the animated open/close element
* `disconnectOpenCloseComponent` – called on `disconnectedCallback` and removes listeners for transition start/end on the animated open/close element
* `onToggleOpenCloseComponent` – called on components not implementing `open` and if duration is 0s, calls before+open/close hooks accordingly and if duration is > 0s, it will add listeners for transition events, which are removed after being invoked.

With a few tweaks (namely proper listener cleanup), I think we could use `onToggleOpenCloseComponent` wherever the open prop is updated and no longer need the other utils.

### Interim solution

If we restore the workaround from and adjust the internal duration prop usage from #5981, we could restore behavior without introducing regressions.

**Comment by jcfranco** [comment]
Installed a fix for `prefers-reduced-motion`. Environments setting `--calcite-duration-factor` to 0 will still have this issue, but using a value of 0.01 can be used as a workaround.

**Comment by geospatialem** [comment]
Verified on `master` when animations are "off" on Windows.

Also verified the animations when `prefers-reduced-motion` is set for `loader`, `block` (with loader), and `progress`.
